prospector/api/main.py

```diff
@@ -89,7 +90,7 @@ async def create_data(repository_url, commit_id, label, vulnerability_id):
 @app.get("/commits/{repository_url}")
 # async def get_commits(repository_url, commit_id=None, token=Depends(oauth2_scheme)):
 async def get_commits(repository_url, commit_id=None):
-    commit = {"commit_id": commit_id, "repository_url": repository_url}
+    commit = Commit(commit_id, repository_url)
```

Comment:
Exactly!
BTW: the type hints and param. names will help, through IDE autocompletion, understand what each parameter is supposed to be.